### ISSUE TITLE
[JSX props] createMemo called when ternary operator used as value for component props

### ISSUE THREAD

**Issue by jakerdy** [solidjs/solid#1801]
### Describe the bug

Full example here: https://playground.solidjs.com/anonymous/3e5096e4-4f46-49c9-8aa2-eebe2a8f17f1

-----------

Hi, there is some weird behaviour related to ternary operators inside jsx property expressions.

When i do thing like this:
```typescript
<DraggableCounter val={mode() ? v1() : v2()} set={setV2} />
```
I got lots of `computations created outside a `createRoot` or `render` will never be disposed`

Context: `<DraggableCounter>` in this case used to tweak values in store, and could work in two modes depending on data that stored in same store. It changes values while user drags in with mouse.

-----------

I checked code that transpiled, and find out that for some reason, my ternary operator has changed from :
```typescript
val={mode() ? v1() : v2()}
```
to:
```tyepscript
get val() {
  return _$memo(() => !!mode())() ? v1() : v2();
}
```
Which is obviously incorrect, because i don't want to create any memo's in this situation, because this property will be accessed multiple times ouside of rendering flow of solid. Also creating memo each time when property got accessed is 'suboptimal' in terms of memory.

-----------

Is that behaviour intentional?

Of course I could create workaround using `()=>number` instad of "plain value" in props object. But it not always the case, cause it could be some "library" code which i couldn't modify.

Also it doesn't feel 'ok' because memo will be created every time I accessing this prop, which elliminates point of using memo in this case. It should be created somewhere else, so that it could persist between getter invokations.

### Steps to Reproduce the Bug or Issue

Pass ternary operator wtih () calls to "value" attribute of component

### Expected behavior

Code should be passed as is without rewriting it's logic. If i need memo, i will make it by my self.

### Screenshots or Videos

First two props depends on other values, and `Left/Right` could become `Pos X/Width`, and source value which was passed to one of number editors was implemented via ternary op, passed to 'value' property of number editor component.

Screen from my project: 
![image](https://github.com/solidjs/solid/assets/66637406/7c7903d3-ea8e-4e19-b38f-30a3d0c77876)

### Platform

- OS: Windows
- Browser: Brave (Chrome)
- Version: Latest

**Comment by ryansolid** [comment]
This is basically a duplicate of #1715. This was intentional because it was difficult to heuristically tell if the work you would be doing in the branch would be expensive. The heuristic to date has been if there is a function call on the other side of the ternary and it depends on a potentially reactive value we wrap.

There is an argument that we could limit this to JSX. In a sense Solid has gotten more strict over time to the degree that we don't expect people to use Functions and Component instantiations interchangeably like we did early days when I first went with this heuristic. But in general I'd recommend reading the response from the linked issue as well.
